### ISSUE TITLE
(Feature request) Wikify a note

### ISSUE THREAD

**Issue by diyoyo** [TriliumNext/Notes#788]
### Describe feature

Similar to "search and replace", I'm thinking of a tool that could wikify a given note, ie transform a plain text into a note that connects other notes together everytime the keyword appears.
Currently, I do it manually, using "Ctrl+L" to trigger the popup menu and link a part of a sentence to a given trilium note.
But what if we could automate the search and link, at least for exact matches ?

Before
<img width="867" alt="image" src="https://github.com/user-attachments/assets/b0de330b-82e5-4eac-abe7-c8c1c4d9da16" />

After
<img width="899" alt="image" src="https://github.com/user-attachments/assets/3f4ffb70-a733-4d64-b010-0554400edb1a" />

### Additional Information

Implementation thought:
- I guess it would go in the sandwich menu on the top right
- it would require taking into account duplicate titles when matching words
- an extra feature, just like "Search and Replace", would be to go "one by one", to confirm the linking more granularly

**Comment by Nriver** [comment]
Interesting idea.

**Comment by diyoyo** [comment]
> No, my understanding was that it would only link to other notes in the Trilium instance. Sounds like a neat idea to me.

That's correct, this is what I meant. @CobriMediaJulien , Sorry for the confusion.



> the "automatic" linking would only work for words/phrases, that are an exact match of title of another note, correct?
> 
> e.g. let's say you have a note with the title `Test Note ABC`
> 
> and you then create a new note, with a content like
> 
> ```
> This is a note, but the best note is Test Note ABC
> ```
> 
> you would want the text part `Test Note ABC` to be linked with the Note, that has the same title, right?

That's correct. As far as I pushed the reasoning yesterday, the non-obvious links will still have to be done manually.

however... I realize I'm already doing a trick, even for manual linking, since the `Meta+L` already suggests that the note is to be searched based on the selected text:
So far, my trick is to have a label `#alias` that is hidden in the attributes, so that the search still returns the correct result (at least in the top list).

Example:
note title : `John Doe`
attribute : #label:alias="Dad"

Maybe the automated linking could try to match the aliases too. Maybe only in a supervised ("Do you confirm (y/n)") way though, since it would be tricky to avoid duplicate issues.

I'm starting to implement aliases for translations as well, since I have days when I write in english and days when I write in french, but I still want to refer to the same note.

**Comment by Nriver** [comment]
I’ve implemented this feature using **trilium-py**. You can find more details in the documentation:  
[Advanced Usage - Automatically Add Internal Links](https://github.com/Nriver/trilium-py?tab=readme-ov-file#advanced-usage--automatically-add-internal-links).

Please note that I haven’t tested it extensively across all my notes. There’s a possibility of errors, which could potentially disrupt your notes. To stay safe, make sure to back up your notes before proceeding.
